app/routers.py

```diff
@@ -398,18 +406,48 @@ async def get_source_tags(
 
     return vectordb.get_available_labels()
 
-@router.get('/api/get-supabase-keys', response_model=schema.SupabaseKeys)
-def get_supabase_keys():
-    """Returns the supabase keys if set in environment variables"""
-    supabase_url = os.environ.get('SUPABASE_URL')
-    supabase_key = os.environ.get('SUPABASE_KEY')
 
-    if not supabase_url or not supabase_key:
-        return {"supabaseUrl": "", "supabaseKey": ""}
+@router.post("/login")
+async def login(
+    email=Form(..., desc="Email of the user"),
+    password=Form(..., desc="Password of the user"),
+    ):
+    """Signs in a user"""
+    supabase_url: str = os.environ.get("SUPABASE_URL")
+    supabase_key: str = os.environ.get("SUPABASE_KEY")
+    supabase: Client = create_client(supabase_url, supabase_key)
+    try:
+        data = supabase.auth.sign_in_with_password({"email": email, "password": password})
+    except gotrue.errors.AuthApiError as e:
+        raise PermissionException("Unauthorized access. Invalid token.") from e
 
-    return {"supabaseUrl": supabase_url, "supabaseKey": supabase_key}
+    return {"access_token": data.session.access_token}
 
-@router.get("/modules/{module_name}")
-async def get_module(module_name: str):
-    """Returns the module file"""
-    return FileResponse(f"modules/{module_name}")
+
+@router.post("/logout")
+async def logout(
+    ):
+    """Signs out a user"""
+    supabase_url: str = os.environ.get("SUPABASE_URL")
+    supabase_key: str = os.environ.get("SUPABASE_KEY")
+    supabase: Client = create_client(supabase_url, supabase_key)
+    supabase.auth.sign_out()
+
+    return {"next_url": f"http://{DOMAIN}/login"}
+
+
+@router.post("/signup")
+async def signup(
+    email=Form(..., desc="Email of the user"),
+    password=Form(..., desc="Password of the user"),
+    ):
+    """Signs up a new user"""
```

Comment:
Do we need more details about the user, like organization?
Where does the access allocation happen? Like which API lets us permit a user to access a label?

Reply:
Yeah, good question. And the organization will need to verify it somehow, because we can't have a user just choosing their organization.
@joelthe1 What do you think?